zebra-state/src/config.rs

```diff
@@ -317,15 +317,6 @@ fn check_and_delete_database(
         return None;
     }
 
-    // Don't delete databases that can be reused.
-    if RESTORABLE_DB_VERSIONS
-        .iter()
-        .map(|v| v - 1)
-        .any(|v| v == dir_major_version)
-    {
-        return None;
-    }
```

Comment:
It was because this function's only called from the start command where we can expect Zebra to rename the directory before it's called, but in hindsight, there's no harm in checking this anyway in case the function is ever used differently.

Reply:
Thanks for the comment. I thought there was some deeper reason I didn't understand.